-mtune=power9 is not working

The makefile doesn't work out of the box on lassen:
```
make: 'lib' with optional backends: /cpu/self/memcheck/serial /cpu/self/memcheck/blocked /cpu/occa /gpu/occa /omp/occa /gpu/cuda/ref /gpu/cuda/reg /gpu/cuda/shared /gpu/cuda/gen
          CC build/interface/ceed-fortran.o
cc1: error: unrecognized argument in option '-mtune=power9'
```
Do you know if there is an easy fix to this?

Here is the output of `make V=1`:
```
[dudouit1@lassen709:libCEED]$ make V=1
make: 'lib' with optional backends: /cpu/self/memcheck/serial /cpu/self/memcheck/blocked /gpu/cuda/ref /gpu/cuda/reg /gpu/cuda/shared /gpu/cuda/gen
gcc -I./include -I/usr/tce/packages/cuda/cuda-10.1.243/include -std=c99 -O -g -mtune=native -ffp-contract=fast -fopenmp-simd -Wall -Wextra -Wno-unused-parameter -fPIC -MMD -MP -DUNDERSCORE  -I/usr/tce/packages/cuda/cuda-10.1.243/include -c -o build/interface/ceed-fortran.o /g/g91/dudouit1/Projects/libCEED/interface/ceed-fortran.c
cc1: error: unrecognized argument in option '-mtune=power9'
cc1: note: valid arguments to '-mtune=' are: 401 403 405 405fp 440 440fp 464 464fp 476 476fp 505 601 602 603 603e 604 604e 620 630 740 7400 7450 750 801 821 823 8540 8548 860 970 G3 G4 G5 a2 cell e300c2 e300c3 e500mc e500mc64 e5500 e6500 ec603e native power3 power4 power5 power5+ power6 power6x power7 power8 powerpc powerpc64 powerpc64le rs64 titan
make: *** [Makefile:371: build/interface/ceed-fortran.o] Error 1
```
I'll try ask some people what is the expected value for `-mtune`

--- a/Makefile
+++ b/Makefile
@@ -76,9 +76,13 @@ AFLAGS = -fsanitize=address #-fsanitize=undefined -fno-omit-frame-pointer
 
 CC_VENDOR := $(firstword $(filter gcc clang,$(shell $(CC) --version)))
 
-MARCHFLAG := $(if $(shell $(CC) -E -march=native -x c /dev/null > /dev/null 2>&1 && echo 1),-march=native,-mtune=native)
+cc_check_flag = $(shell $(CC) -E $(1) -x c /dev/null > /dev/null 2>&1 && echo 1)
+MARCHFLAG := -march=native
+MARCHFLAG := $(if $(call cc_check_flag,$(MARCHFLAG)),$(MARCHFLAG),-mcpu=native)
+MARCHFLAG := $(if $(call cc_check_flag,$(MARCHFLAG)),$(MARCHFLAG))
 
-OMP_SIMD_FLAG := $(if $(shell $(CC) -E -fopenmp-simd -x c /dev/null > /dev/null 2>&1 && echo 1),-fopenmp-simd)
+OMP_SIMD_FLAG := -fopenmp-simd
+OMP_SIMD_FLAG := $(if $(call cc_check_flag,$(OMP_SIMD_FLAG)),$(OMP_SIMD_FLAG))
 
 OPT    ?= -O -g $(MARCHFLAG) -ffp-contract=fast $(OMP_SIMD_FLAG)
 CFLAGS ?= -std=c99 $(OPT) -Wall -Wextra -Wno-unused-parameter -fPIC -MMD -MP
@@ -407,12 +411,12 @@ $(OBJDIR)/nek-bps : examples/nek/bps/bps.usr examples/nek/nek-examples.sh $(libc
 
 $(OBJDIR)/petsc-% : examples/petsc/%.c $(libceed) $(ceed.pc) | $$(@D)/.DIR
 	+$(MAKE) -C examples/petsc CEED_DIR=`pwd` \
-	  PETSC_DIR="$(abspath $(PETSC_DIR))" $*
+	  PETSC_DIR="$(abspath $(PETSC_DIR))" OPT="$(OPT)" $*
 	mv examples/petsc/$* $@
 
 $(OBJDIR)/ns-% : examples/navier-stokes/%.c $(libceed) $(ceed.pc) | $$(@D)/.DIR
 	+$(MAKE) -C examples/navier-stokes CEED_DIR=`pwd` \
-	  PETSC_DIR="$(abspath $(PETSC_DIR))" $*
+	  PETSC_DIR="$(abspath $(PETSC_DIR))" OPT="$(OPT)" $*
 	mv examples/navier-stokes/$* $@
 
 libceed_test.o = $(test_backends.c:%.c=$(OBJDIR)/%.o)

--- a/examples/navier-stokes/Makefile
+++ b/examples/navier-stokes/Makefile
@@ -14,9 +14,6 @@
 # software, applications, hardware, advanced system engineering and early
 # testbed platforms, in support of the nation's exascale computing imperative.
 
-MARCHFLAG := $(if $(shell $(CC) -E -march=native -x c /dev/null > /dev/null 2>&1 && echo 1),-march=native,-mtune=native)
-OPT ?= -O -g $(MARCHFLAG) -ffp-contract=fast -fopenmp-simd
-
 PETSc.pc := $(PETSC_DIR)/$(PETSC_ARCH)/lib/pkgconfig/PETSc.pc
 CEED_DIR ?= ../..
 ceed.pc := $(CEED_DIR)/lib/pkgconfig/ceed.pc

